hilti/runtime/src/library.cc

```diff
@@ -119,6 +119,17 @@ hilti::rt::Result<Nothing> hilti::rt::Library::remove() const {
 
 hilti::rt::Result<hilti::rt::Nothing> hilti::rt::Library::save(const hilti::rt::filesystem::path& path) const {
     std::error_code ec;
+
+    try {
+        // On macOS ARM, there'll be weird crashes during execution if we don't
+        // remove an existing file first. Note that is_regular_file() throws if
+        // we can't access the file due to permissions.
+        if ( hilti::rt::filesystem::is_regular_file(path) )
+            hilti::rt::filesystem::remove(path);
+    } catch ( const hilti::rt::filesystem::filesystem_error& e ) {
+        return result::Error(fmt("could not remove existing library when saving to %s: %s", path, e.what()));
+    }
```

Comment:
Alternative style to make the control flow slightly more explicit:

```suggestion
    // On macOS ARM, there are weird crashes during execution if we don't remove an existing file first. Note that
    // `is_regular_file(`) fails if we cannot access the file due to permissions, but we let it fail during removal.
    if ( hilti::rt::filesystem::is_regular_file(path, ec) ) {
        hilti::rt::filesystem::remove(path, ec);
        if ( ec )
            return result::Error(fmt("could not remove existing library when saving to %s: %s", path, ec.message()));
    }
```

Reply:
Ah, I had missed that there's an `ec` version of `is_regular_file()` as well.